packages/oidc-dpop-client-browser/src/dpop/dpop.ts

```diff
@@ -30,14 +43,111 @@ import { JWK } from "node-jose";
  * @hidden
  */
 export async function generateJWK(
-  kty: "EC" | "OKP" | "RSA" | "oct",
+  kty: "EC" | "RSA",
   crvBitlength?: ECCurve | OKPCurve | number,
   parameters?: BasicParameters
 ): Promise<JSONWebKey> {
   const key = await JWK.createKey(kty, crvBitlength, parameters);
   return key.toJSON(true) as JSONWebKey;
 }
 
+/**
+ * Generates a Json Web Token (https://tools.ietf.org/html/rfc7519) containing
+ * the provided payload and using the signature algorithm specified in the options.
+ * @param payload The body of the JWT.
+ * @param key The key used for the signature.
+ * @param options
+ * @returns a 3-parts base64-encoded string, split by dots.
+ * @hidden
+ */
+export async function signJWT(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  payload: Record<string, any>,
+  key: JWKECKey | JWKOKPKey | JWKRSAKey | JWKOctKey,
+  options?: JoseJWT.SignOptions
+): Promise<string> {
+  const parsedKey = await JWK.asKey(key);
+  const convertedKey: string = parsedKey.toPEM(true);
+  const signed = JWT.sign(payload, convertedKey, {
+    ...(options as JWT.SignOptions),
+  });
+  return signed;
+}
+
+/**
+ * Decodes the base64 Json Web Token into an object. If a key is specified, the
+ * JWT is also verified.
+ * @param token The base64-encoded token
+ * @param key The key used to sign the token
+ * @returns the payload of the JWT
+ * @hidden
+ */
+export async function decodeJWT(
+  token: string,
+  key?: JWKECKey | JWKOKPKey | JWKRSAKey | JWKOctKey,
+  options?: VerifyOptions
+): Promise<Record<string, unknown>> {
+  if (key) {
+    const parsedKey = await JWK.asKey(key);
+    const convertedKey: string = parsedKey.toPEM(false);
+    return JWT.verify(token, convertedKey, options) as Promise<
+      Record<string, unknown>
+    >;
+  }
+  return JWT.decode(token) as Promise<Record<string, unknown>>;
+}
+
+/**
+ * @param key
+ * @hidden
+ */
+export async function privateJWKToPublicJWK(
+  key: JSONWebKey
+): Promise<JSONWebKey> {
+  return (await JWK.asKey(key as JWK.RawKey, "public")) as JSONWebKey;
+}
+
+/**
+ * Normalizes a URL in order to generate the DPoP token based on a consistent scheme.
+ * @param audience The URL to normalize.
+ * @returns The normalized URL as a string.
+ * @hidden
+ */
+export function normalizeHtu(audience: URL): string {
```

Comment:
Can we rename this to English, e.g. `normalizeHttpUriClaim()` (I don't know if `htu` is technically a 'claim')? Unfortunately this function isn't inside a Class to help give context, but is it really DPoP-specific?

Reply:
Changed in eb02277a9b53f21ce24688d549a5908ec816b2ac. I don't really think there is a need to specify that this is DPoP specific, as the function is only ever used in the `dpop.ts` file, and it is exported only for test purpose.

Reply:
Yeah, my point was just that if it's **_not_** DPoP-specific then should it be moved to a Utils class, or a Claims class, or something. But it's perfectly fine here - I'd only consider moving it (perhaps into the Core package) if something **_else_** wanted to call this function too.